File: DS3Lib/Sources/DS3Lib/Models/Account.swift
/// Account email model
struct AccountEmail: Codable {
    var id: String
    var email: String
    
    /// Whether this is the default email for the account
    var isDefault: Bool
    
    var createdAt: String
    var isVerified: Bool
    var tenantId: String
    
    private enum CodingKeys: String, CodingKey {
        case id
        case email
        case isDefault = "default"
        case createdAt = "created_at"
        case isVerified = "verified"
        case tenantId = "tenant_id"
    }
}

/// An account in the Cubbit's DS3 ecosystem
struct Account: Codable {
    var id: String
    var firstName: String
    var lastName: String
    var isInternal: Bool
    var isBanned: Bool
    var createdAt: String
    var deletedAt: String?
    var bannedAt: String?
    var maxAllowedProjects: Int32
    var emails: [AccountEmail]
    var isTwoFactorEnabled: Bool
    var tenantId: String
    var endpointGateway: String
    var authProvider: String
    
    private enum CodingKeys: String, CodingKey {
        case id
        case firstName = "first_name"
        case lastName = "last_name"
        case isInternal = "internal"
        case isBanned = "banned"
        case createdAt = "created_at"
        case deletedAt = "deleted_at"
        case bannedAt = "banned_at"
        case maxAllowedProjects = "max_allowed_projects"
        case emails
        case isTwoFactorEnabled = "two_factor_enabled"
        case tenantId = "tenant_id"
        case endpointGateway = "endpoint_gateway"
        case authProvider = "auth_provider"
    }
}

File: core/ds3-models/src/account.rs
//! Account and AccountEmail types for the Cubbit DS3 platform.

use serde::{Deserialize, Serialize};

/// An email address associated with a Cubbit account.
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub struct AccountEmail {
    /// The email record identifier.
    pub id: String,

    /// The email address.
    pub email: String,

    /// Whether this is the default email for the account.
    #[serde(rename = "default")]
    pub is_default: bool,

    /// When the email was created (ISO 8601).
    #[serde(rename = "created_at")]
    pub created_at: String,

    /// Whether the email has been verified.
    #[serde(rename = "verified")]
    pub is_verified: bool,

    /// The tenant identifier.
    #[serde(rename = "tenant_id")]
    pub tenant_id: String,
}

/// An account in the Cubbit DS3 ecosystem.
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub struct Account {
    /// The account unique identifier.
    pub id: String,

    /// The user's first name.
    #[serde(rename = "first_name")]
    pub first_name: String,

    /// The user's last name.
    #[serde(rename = "last_name")]
    pub last_name: String,

    /// Whether the account is internal.
    #[serde(rename = "internal")]
    pub is_internal: bool,

    /// Whether the account is banned.
    #[serde(rename = "banned")]
    pub is_banned: bool,

    /// When the account was created (ISO 8601).
    #[serde(rename = "created_at")]
    pub created_at: String,

    /// When the account was deleted, if applicable.
    #[serde(rename = "deleted_at")]
    pub deleted_at: Option<String>,

    /// When the account was banned, if applicable.
    #[serde(rename = "banned_at")]
    pub banned_at: Option<String>,

    /// Maximum number of projects allowed.
    #[serde(rename = "max_allowed_projects")]
    pub max_allowed_projects: i32,

    /// The account's email addresses.
    pub emails: Vec<AccountEmail>,

    /// Whether two-factor authentication is enabled.
    #[serde(rename = "two_factor_enabled")]
    pub is_two_factor_enabled: bool,

    /// The tenant identifier.
    #[serde(rename = "tenant_id")]
    pub tenant_id: String,

    /// The S3 endpoint gateway URL.
    #[serde(rename = "endpoint_gateway")]
    pub endpoint_gateway: String,

    /// The authentication provider.
    #[serde(rename = "auth_provider")]
    pub auth_provider: String,
}
